api/internal/crawl/crawler/crawler.go

```diff
@@ -102,36 +102,39 @@ func CrawlFromSeed(ctx context.Context, seed CrawlSeed,
 	}
 
 	doCrawl := func(docsPtr *CrawlSeed) {
-		for len(*docsPtr) > 0 {
-			back := len(*docsPtr) - 1
-			next := (*docsPtr)[back]
-			*docsPtr = (*docsPtr)[:back]
-
+		n := len(*docsPtr)
+		for i := 0; i < n; i++ {
+			next := (*docsPtr)[i]
 			match := findMatch(next)
 			if match == nil {
 				logIfErr(fmt.Errorf(
 					"%v could not match any crawler", next))
 				continue
 			}
 
+			fmt.Println("Crawling ", next.RepositoryURL, next.FilePath)
```

Comment:
please use logger instead (let's be consistent)

Reply:
Done